catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/SeedCommand.java

```diff
@@ -0,0 +1,241 @@
+
+            final QueryRequestImpl queryRequest;
+            if (isNotEmpty(sources)) {
+                queryRequest = new QueryRequestImpl(query, sources);
+            } else {
+                queryRequest = new QueryRequestImpl(query, true);
+            }
+            queryRequest.setProperties(new HashMap<>(CACHE_UPDATE_PROPERTIES));
+
+            final QueryResponse queryResponse = framework.query(queryRequest);
+
+            if (queryResponse.getResults()
+                    .isEmpty()) {
+                break;
+            }
+
+            final List<Entry<? extends ResourceRequest, String>> resourceRequests =
+                    queryResponse.getResults()
+                            .stream()
+                            .map(Result::getMetacard)
+                            .filter(isNonCachedResource)
+                            .map(metacard -> new SimpleEntry<>(new ResourceRequestById(metacard.getId()),
+                                    metacard.getSourceId()))
+                            .collect(toList());
+
+            for (Entry<? extends ResourceRequest, String> requestAndSourceId : resourceRequests) {
+                final ResourceRequest request = requestAndSourceId.getKey();
+                try {
+                    ResourceResponse response = framework.getResource(request,
+                            requestAndSourceId.getValue());
+                    ++resourceDownloads;
+                    EXECUTOR.execute(new ResourceCloseHandler(response));
+                } catch (IOException | ResourceNotFoundException | ResourceNotSupportedException e) {
+                    ++downloadErrors;
+                    LOGGER.warn("Could not download resource for metacard [id={}]",
+                            request.getAttributeValue(),
+                            e);
+                }
+
+                printProgressAndFlush(start, resourceLimit, resourceDownloads);
+
+                if (resourceDownloads == resourceLimit) {
+                    break;
+                }
+            }
+        }
+
+        printProgressAndFlush(start, resourceDownloads, resourceDownloads);
+
+        console.println();
+        if (downloadErrors > 0) {
+            printErrorMessage(downloadErrors
+                    + " resource download(s) had errors. Check the logs for details.");
+        }
+        printSuccessMessage(
+                "Done seeding. " + resourceDownloads + " resource download(s) started.");
+
+        return null;
+    }
+
+    private static class ResourceCloseHandler implements Runnable {
+        private final InputStream resourceStream;
+
+        private ResourceCloseHandler(ResourceResponse response) {
+            resourceStream = response.getResource()
+                    .getInputStream();
+        }
+
+        @Override
+        public void run() {
+            byte[] buffer = new byte[4096];
+            try {
+                while (resourceStream.read(buffer, 0, buffer.length) != -1) {
```

Comment:
You're just reading the stream until its end, dropping its contents on the floor? Because the caching of the metacard/resource is happening in another thread and all you care about is the completion of this stream? Am I understanding that correctly? If so, I take back what I said earlier about the size of the execpool as it's probably okay to be that large to close these out quickly.